pkg/apiclient/client_factory.go

```diff
@@ -82,26 +99,18 @@ func NewClientFactory(httpClient *http.Client, host string, apiKey string, space
 		ApiKey:            apiKey,
 		SpaceNameOrID:     spaceNameOrID,
 		ActiveSpace:       nil,
-		Ask:               asker,
+		Ask:               ask,
 	}
 	return clientImpl, nil
 }
 
 // NewClientFactoryFromEnvironment Creates a new Client wrapper structure by reading the environment.
 // specifies nil for the HTTP Client, so this is not for unit tests; use NewClientFactory(... instead)
-func NewClientFactoryFromEnvironment() (ClientFactory, error) {
+func NewClientFactoryFromEnvironment(ask question.AskProvider) (ClientFactory, error) {
 	host := os.Getenv("OCTOPUS_HOST")
 	apiKey := os.Getenv("OCTOPUS_API_KEY")
 	spaceNameOrID := os.Getenv("OCTOPUS_SPACE")
```

Comment:
We should provide turn these into constants.

Reply:
will do